src/JsonSchema/Constraints/Constraint.php

```diff
@@ -28,8 +28,9 @@
     protected $errors = array();
     protected $inlineSchemaProperty = '$schema';
 
-    const CHECK_MODE_NORMAL = 1;
-    const CHECK_MODE_TYPE_CAST = 2;
+    const CHECK_MODE_NORMAL = 0x00000001;
```

Comment:
I'm fine with it but I would caution that web developers, in my experience, are not usually accustomed to bitwise operations or thinking.

 would go with a boolean literal if that is the clarity you intend to convey:

```php
0b0000001 // 1
0b0000010 // 2
0b0000100 // 4
```